CONTRIBUTING.md

```diff
@@ -384,39 +396,58 @@ Here is a skeleton predicate script:
 # * we access any undefined variable.
 set -eu
 
-# Print out Rust backtraces on panic. Useful for minimizing a particular panic.
-export RUST_BACKTRACE=1
+# Invoke the general purpose predicate script that comes in the
+# `bindgen` repository.
+path/to/rust-bindgen/csmith-fuzzing/predicate.py \
+    $PREDICATE_FLAGS \
+    ./isolated-test-case.h
+```
+
+When hunting down a particular panic emanating from inside `bindgen`, you can
+invoke `predicate.py` like this:
 
-# If the `libclang.so` you're using for `bindgen` isn't the system
-# `libclang.so`, let the linker find it.
-export LD_LIBRARY_PATH=~/path/to/your/directory/containing/libclang
+```bash
+path/to/rust-bindgen/csmith-fuzzing/predicate.py \
+    --expect-bindgen-fail \
+    --bindgen-grep "thread main panicked at '<insert panic message here>'" \
+    ./isolated-test-case.h
+```
 
-# Make sure that the reduced test case is valid C/C++ by compiling it. If it
-# isn't valid C/C++, this command will exit with a nonzero exit code and cause
-# the whole script to do the same.
-clang[++ --std=c++14] -c ./pre_processed_header.hpp
+Alternatively, when hunting down a bad `#[derive(Eq)]` that is causing `rustc`
+to fail to compile `bindgen`'s emitted bindings, you can invoke `predicate.py`
+like this:
 
-# Run `bindgen` and `grep` for the thing your hunting down! Make sure to include
-# `2>&1` to get at stderr if you're hunting down a panic.
-~/src/rust-bindgen/target/debug/bindgen \
-    ./pre_processed_header.hpp \
-    [ <extra flags> ] \
-    2>&1 \
-    | grep "<pattern in generated bindings or a panic string or ...>"
+```bash
+path/to/rust-bindgen/csmith-fuzzing/predicate.py \
+    --bindings-grep NameOfTheStructThatIsErroneouslyDerivingEq \
+    --expect-compile-fail \
+    --rustc-grep 'error[E0277]: the trait bound `f64: std::cmp::Eq` is not satisfied' \
+    ./isolated-test-case.h
 ```
 
-When hunting down a panic, I `grep`ed like this:
+Or, when minimizing a failing layout test in the compiled bindings, you can
+invoke `predicate.py` like this:
 
-    ... | grep "thread main panicked at '<panic error message here>'"
+```bash
+path/to/rust-bindgen/csmith-fuzzing/predicate.py \
+    --bindings-grep MyStruct \
+    --expect-layout-tests-fail \
+    --layout-tests-grep "thread 'bindgen_test_layout_MyStruct' panicked" \
+    ./isolated-test-case.h
+```
+
+For details on all the flags that you can pass to `predicate.py`, run:
 
-When hunting down bad codegen for a base member, I `grep`ed like this:
+```
+$ path/to/rust-bindgen/csmith-fuzzing/predicate.py --help
+```
 
-    ... | grep "pub _base: MyInvalidBaseTypeThatShouldntBeHere"
+And you can always write your own, arbitrary predicate script if you prefer.
```

Comment:
Maybe we should provide a link to the old hand-written predicate script example? 
I think it may be very helpful for those who stumbled on lack of `predicate.py` functionality. Before extending `predicate.py` one might want to do experiments which may be more productive with bare shell scripts.